### ISSUE TITLE
check_syntax: false positive for `AT SELECTION-SCREEN`

### ISSUE THREAD

**Issue by mbtools** [abaplint/abaplint#3816]
I'm getting `check_syntax: Variable name "test" already defined` error which is not correct. This is allowed 😉 

```abap
REPORT zabaplint_var_exits2.

AT SELECTION-SCREEN.

  DATA(test) = 1.

START-OF-SELECTION.

  DATA(test) = 2.
```

Note: If you replace `AT SELECTION-SCREEN` with `INITIALIZATION` or `AT LINE-SELECTION`, then the error is correct.

<img width="1012" height="119" alt="Image" src="https://github.com/user-attachments/assets/99f85822-9c29-4110-9c4c-1a4f79b480ac" />

https://help.sap.com/doc/abapdocu_latest_index_htm/latest/en-US/ABAPAT_SELECTION-SCREEN.html

**Comment by larshp** [comment]
another example,

```abap
DATA gv_class TYPE string.

AT SELECTION-SCREEN OUTPUT.
  DATA gv_class TYPE i.
```

and an example where an error is expected:

```abap
AT SELECTION-SCREEN.
  DATA(test) = 1.

START-OF-SELECTION.
  WRITE / test.
```

however this is okay,
```abap
START-OF-SELECTION.
  DATA(test) = 1.

FORM foo.
  WRITE / test.
ENDFORM.
```

**Comment by larshp** [comment]
@mbtools try again, 2.119.3

**Comment by mbtools** [comment]
looks good. thx
